test/built-ins/Proxy/defineProperty/targetdesc-not-configurable-writable-desc-not-writable.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2019 Aleksey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 9.5.6
+description: >
+    Throw a TypeError exception if trap result is true, targetDesc is not configurable
+    and writable, while Desc is not writable.
+info: |
+    [[DefineOwnProperty]] (P, Desc)
+
+    ...
+    16. Else targetDesc is not undefined,
+        ...
+        c. If IsDataDescriptor(targetDesc) is true, targetDesc.[[Configurable]] is
+        false, and targetDesc.[[Writable]] is true, then
+            i. If Desc has a [[Writable]] field and Desc.[[Writable]] is
+            false, throw a TypeError exception.
+    ...
+features: [Proxy]
+---*/
+
+var target = {};
+var p = new Proxy(target, {
+  defineProperty: function(t, prop, desc) {
+    return true;
+  },
+});
+
+Object.defineProperty(target, "prop", {
+  configurable: false,
+  writable: true,
+});
+
+assert.throws(TypeError, function() {
+  Object.defineProperty(p, "prop", {
```

Comment:
Use Reflect to avoid false positives from [`DefinePropertyOrThrow`](https://tc39.github.io/ecma262/#sec-definepropertyorthrow), this way we assert the internal throws the TypeError and do not return `false`.

Reply:
63026bb